### PR TITLE
Add "Actions" docs section under NLU-based assistants

### PR DESCRIPTION
**Pull request by m-vdb**
**Proposed changes**:
- Add "Actions" docs section under NLU-based assistants, with:
  - overview
  - default actions
  - slot validation actions (moved from `concepts/`)
- Update `concepts/default-actions.mdx` content

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: docs/docs/action-server/actions.mdx [hunk 1]
```diff
@@ -308,7 +308,7 @@ and return both messages to the user.
 ## Special Action Types
 
 There are special action types that are automatically triggered under certain circumstances, namely [default actions](../concepts/default-actions.mdx)
-and [slot validation actions](../concepts/slot-validation-actions.mdx).
+and [slot validation actions](../nlu-based-assistants/slot-validation-actions.mdx).
```
Comment (amn41) [inline docs/docs/action-server/actions.mdx:311]: @dakshvar22 , we are still using slot validation actions in calm, aren't we? if you have custom logic for how to accept / reject a slot value

File: docs/docs/concepts/default-actions.mdx [hunk 2]
```diff
@@ -135,128 +135,3 @@ class ActionSessionStart(Action):
         # as a user message follows
         return [SessionStarted(), ActionExecuted("action_listen")]
 ```
-
-## action_default_fallback
-
-This action undoes the last user-bot interaction and sends the `utter_default` response if it is defined.
```
Comment (amn41) [inline docs/docs/concepts/default-actions.mdx:141]: @dakshvar22 we do still have a fallback, no? I thought I saw a ticket for handling this in docsearch

**Review by amn41 — CHANGES_REQUESTED** [review]
tagged Daksh on a couple of product questions that need clarification

File: docs/docs/action-server/actions.mdx (hunk 1 above)
Comment (dakshvar22) [reply docs/docs/action-server/actions.mdx:311]: Yes, slot validation actions are still used.

File: docs/docs/concepts/default-actions.mdx (hunk 2 above)
Comment (dakshvar22) [reply docs/docs/concepts/default-actions.mdx:141]: Yes, we plan to have it. I also understood today from Thomas that it is not being triggered anywhere yet but it needs to in specific scenarios.

Comment (m-vdb) [reply docs/docs/concepts/default-actions.mdx:141]: the current docs for this currently refer to the [fallback-handhoff page](https://rasa.com/docs/rasa/fallback-handoff) which is currently written in the context of NLU-based assistants. @dakshvar22 can you make sure that when this is implemented, the docs are updated to reflect the way it works with CALM?

File: docs/docs/action-server/actions.mdx (hunk 1 above)
Comment (ancalita) [reply docs/docs/action-server/actions.mdx:311]: Yes, but only regular actions can be used and we also expect these actions now to be named `validate_<slot_name>` (I made a note of that under Flows `collect_information` step) and the rasa-sdk helper actions only work with nlu-based assistants, so this change is accurate.

File: docs/docs/concepts/default-actions.mdx (hunk 2 above)
Comment (dakshvar22) [reply docs/docs/concepts/default-actions.mdx:141]: yep, will take it as part of this ticket - https://rasahq.atlassian.net/browse/ENG-579

**Review by amn41 — APPROVED** [review]

**Pull request merged**